lib/tasks/one_off.rake

```diff
@@ -118,9 +118,59 @@ namespace :one_off do
     puts "Updating #{projects.count} projects..."
 
     projects.in_batches(of: 10000).each do |batch|
-      batch.update_all('status_checked_at = updated_at')
+      batch.update_all("status_checked_at = updated_at")
     end
 
     print "Finished updating projects"
   end
+
+  desc "Backfill Pypi project dependencies kind with environment markers"
+  task :backfill_pypi_dependencies_kind, %i[batch_size] => :environment do |_t, args|
+    # There's ~5 million Pypi versions
+    # Based on a sampling of 5000 Pypi versions, ~1/10 contain environment markers
+    # 4 Pypi API calls per second = 240/min
+    # leaving a buffer of 40 for any batches which contain more, 200/min is conservative.
+    # 200/min = 12000/hour = 288k/day
+    # At that rate it will take ~1.75 days
+
+    pypi_versions = Version.where(
+      project_id: Project.where(platform: "Pypi")
+    )
```

Comment:
Do you know what SQL is getting generated for this? Is it using an `IN` clause with 5 million IDs?